libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -540,6 +540,13 @@ bool AP_BattMonitor::capacity_remaining_pct(uint8_t &percentage, uint8_t instanc
     return false;
 }
 
+/// capacity_remaining_pct_mavlink - returns battery percentage if percentage is valid, -1 otherwise
+int8_t AP_BattMonitor::capacity_remaining_pct_mavlink(uint8_t instance) const
+{
+    uint8_t percentage;
+    return (this->capacity_remaining_pct(percentage, instance)) ? MIN(percentage, INT8_MAX) : -1;
```

Comment:
I'm not sure if everyone is going to be fans of having this specific function for capacity_remaining. Usually, we just let each caller do this bit.  I'm in the not a fan camp.

Anyways just for c++ stuff you should be able to do this, 
```suggestion
    return capacity_remaining_pct(percentage, instance) ? MIN(percentage, INT8_MAX) : -1;
```

Reply:
@hendjoshsr71  I understand. But `capacity_remaining_pct` works on `uint8_t` and in mavlink we need to take care of -1. Storing -1 in `uint8_t` and then assigning or typecasting it to `int8_t` looks ugly.

Reply:
Oh I totally agree about the casting for the `uint8 = -1` :)

That is why I suggested one possible alternative below that seemed simple and doesn't require additional functions.

Reply:
Yes but there are two more places in GCS_Common where we would have to do the same thing. I guess writing a dedicated method would look more cleaner instead of repeating same thing at multiple places.

Reply:
Reasonable enough.